packages/core/data-transfer/lib/engine/__tests__/engine.test.ts

```diff
@@ -0,0 +1,390 @@
+import { ObjectWritableMock, ObjectReadableMock } from 'stream-mock';
+import { createTransferEngine } from '..';
+import {
+  IDestinationProvider,
+  ISourceProvider,
+  ITransferEngine,
+  ITransferEngineOptions,
+} from '../../../types';
+
+const providerStages = ['bootstrap', 'close'];
+
+const getMockSourceStream = (data: Iterable<any> = ['foo', 'bar']) => {
+  const stream = new ObjectReadableMock(data).on('close', () => {
+    stream.destroy();
+  });
+
+  return stream;
+};
+
+const getMockDestinationStream = () => {
+  const stream = new ObjectWritableMock().on('close', () => {
+    stream.destroy();
+  });
+  return stream;
+};
+
+const sourceStages = [
+  ...providerStages,
+  'streamEntities',
+  'streamLinks',
+  // 'streamMedia',
+  'streamConfiguration',
+  'streamSchemas',
+];
+
+const destinationStages = [
+  ...providerStages,
+  'getEntitiesStream',
+  'getLinksStream',
+  // 'getMediaStream',
+  'getConfigurationStream',
+  'getSchemasStream',
+];
+
+// add some helpers to jest
+expect.extend({
```

Comment:
When using extend in a test file context, it only use the extensions for the current file, right?